sql/src/main/java/io/crate/protocols/postgres/AuthenticationMethod.java

```diff
@@ -0,0 +1,30 @@
+package io.crate.protocols.postgres;
+
+
+import org.elasticsearch.common.settings.Settings;
+import org.jboss.netty.channel.Channel;
+
+/**
+ * Common interface for Authentication methods.
+ *
+ * An auth method must provide a unique name which is exposed via the {@link #name()} method.
+ *
+ * It is also responsible for authentication for the Postgres Wire Protocol,
+ * {@link #pgAuthenticate(Channel channel, Settings settings)},
+ */
+public interface AuthenticationMethod {
+    /**
+     * Authenticates the Postgres Wire Protocol client,
+     * sends AuthenticationOK if authentication is successful
+     * If authentication fails it send ErrorResponse
```

Comment:
It sounds to me that every implementation must do this - so it's a method contract - which at this level is not implementation specific. Please keep it, otherwise if you want to implement the method you've no idea what you should do